test/js/upload-wizard/upload-new-study.test.js

```diff
@@ -1,60 +1,48 @@
-import React from 'react'
-import { render, screen, fireEvent, waitForElementToBeRemoved } from '@testing-library/react'
+import { screen, fireEvent, waitForElementToBeRemoved } from '@testing-library/react'
 import '@testing-library/jest-dom/extend-expect'
 import _cloneDeep from 'lodash/cloneDeep'
 import selectEvent from 'react-select-event'
-import ReactNotification from 'react-notifications-component'
 
-import { RawUploadWizard } from 'components/upload/UploadWizard'
-import MockRouter from '../lib/MockRouter'
 import { fireFileSelectionEvent } from '../lib/file-mock-utils'
 import * as ScpApi from 'lib/scp-api'
 import {
-  EMPTY_STUDY, RAW_COUNTS_FILE, PROCESSED_MATRIX_FILE, METADATA_FILE,
-  CLUSTER_FILE
+  RAW_COUNTS_FILE, PROCESSED_MATRIX_FILE, METADATA_FILE,
+  CLUSTER_FILE, COORDINATE_LABEL_FILE, FASTQ_FILE
 } from './file-info-responses'
-import { UserContext } from 'providers/UserProvider'
-
-/** gets a pointer to the react-select node based on label text
- * This is non-trivial since our labels contain the select,
- * and so a naive getByLabelText will not work.
- * Instead we get the label using getByText and assume
- * the first div inside is the react-select element */
-function getSelectByLabelText(screen, text) {
-  return screen.getByText(text).querySelector('div')
-}
+import { renderWizardWithStudy, getSelectByLabelText, saveButton } from './upload-wizard-test-utils'
+
 
 describe('creation of study files', () => {
-  it('renders the raw counts form and allows updates and save', async () => {
-    const studyInfoSpy = jest.spyOn(ScpApi, 'fetchStudyFileInfo')
-    // pass in a clone of the response since it may get modified by the cache operations
-    studyInfoSpy.mockImplementation(params => {
-      const response = _cloneDeep(EMPTY_STUDY)
-      return Promise.resolve(response)
-    })
+  beforeAll(() => {
+    jest.restoreAllMocks()
+    // This test is long, so allow extra time
+    jest.setTimeout(30000)
```

Comment:
Blocker: 30 seconds is > 10x too long.  Running all JS tests (`time yarn test`) without this just took me 28 seconds, not including this.  This would let that double, which is a developer-experience bug.

If fixing that DX bug is infeasible (see other more specific Blocker comment), then please at least add more detail on why these tests are so long in a code comment here.  Maybe also move them to another directory, or otherwise give an easy hook to exclude them so running big batches of local JS tests doesn't take twice as long by default.  (We'd still always run them in CI, of course.)

Reply:
Good point, I was just noticing occasional timeouts at 5 seconds, and I went very conservative so as to avoid sporadic fails.  I'll drop it to 10 seconds.

Reply:
I also think that we should be prepared for our front-end testing to eventually take almost as long as our backend testing -- Right now our frontend coverage is somewhat minimal, and as you note, it still takes 30 seconds.  I like your later comment about looking to parallelize these

Reply:
Thinking more about this, I realize that I wasn't really doing any explicit testing of the wizard state over time to benefit from the serialized form.  So I've added more explicit checks after everything is uploaded

Reply:
Great comments -- that works.  I've suggested a quick comment that would help.